service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/managedIdentity/CreateAzureManagedIdentityStep.java

```diff
@@ -0,0 +1,110 @@
+package bio.terra.workspace.service.resource.controlled.cloud.azure.managedIdentity;
+
+import bio.terra.cloudres.azure.resourcemanager.common.Defaults;
+import bio.terra.cloudres.azure.resourcemanager.msi.data.CreateUserAssignedManagedIdentityRequestData;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.app.configuration.external.AzureConfiguration;
+import bio.terra.workspace.common.exception.AzureManagementExceptionUtils;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import com.azure.core.management.Region;
+import com.azure.core.management.exception.ManagementException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Creates an Azure Managed Identity. Designed to run directly after {@link
+ * GetAzureManagedIdentityStep}.
+ */
+public class CreateAzureManagedIdentityStep implements Step {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(CreateAzureManagedIdentityStep.class);
+  private final AzureConfiguration azureConfig;
+  private final CrlService crlService;
+  private final ControlledAzureManagedIdentityResource resource;
+
+  public CreateAzureManagedIdentityStep(
+      AzureConfiguration azureConfig,
+      CrlService crlService,
+      ControlledAzureManagedIdentityResource resource) {
+    this.azureConfig = azureConfig;
+    this.crlService = crlService;
+    this.resource = resource;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    final AzureCloudContext azureCloudContext =
+        context
+            .getWorkingMap()
+            .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
+    var msiManager = crlService.getMsiManager(azureCloudContext, azureConfig);
+
+    try {
+      msiManager
+          .identities()
+          .define(resource.getManagedIdentityName())
+          .withRegion(resource.getRegion())
+          .withExistingResourceGroup(azureCloudContext.getAzureResourceGroupId())
+          .withTag("workspaceId", resource.getWorkspaceId().toString())
+          .withTag("resourceId", resource.getResourceId().toString())
+          .create(
+              Defaults.buildContext(
+                  CreateUserAssignedManagedIdentityRequestData.builder()
+                      .setName(resource.getManagedIdentityName())
+                      .setRegion(Region.fromName(resource.getRegion()))
```

Comment:
It came up before that the `Region` enum in the Java SDK is incomplete. Most of these APIs have a `setRegion(String)` instead and throw a sensible 400 on errors.

See: https://github.com/DataBiosphere/terra-workspace-manager/pull/1220

Reply:
this is actually a problem in CRL. This region value is just in the context, not actually used for creation, only logging I think

Reply:
oh I missed this was a CRL thing